lib/lbt/utils/JSTokenizer.js

```diff
@@ -0,0 +1,381 @@
+			this.next('a');
+			this.next('l');
+			this.next('s');
+			this.next('e');
+			return false;
+		case 'n':
+			this.next('n');
+			this.next('u');
+			this.next('l');
+			this.next('l');
+			return null;
+		}
+		this.error("Unexpected '" + this.ch + "'");
+	};
+
+		//value, // Place holder for the value function.
+	JSTokenizer.prototype.array = function() {
+
+		// Parse an array value.
+		var array = [];
+
+		if (this.ch === '[') {
+			this.next('[');
+			this.white();
+			if (this.ch === ']') {
+				this.next(']');
+				return array; // empty array
+			}
+			while (this.ch) {
+				array.push(this.value());
+				this.white();
+				if (this.ch === ']') {
+					this.next(']');
+					return array;
+				}
+				this.next(',');
+				this.white();
+			}
+		}
+		this.error("Bad array");
+	};
+
+	var object = function() {
+
+		// Parse an object value.
+		var key, object = {};
+
+		if (this.ch === '{') {
+			this.next('{');
+			this.white();
+			if (this.ch === '}') {
+				this.next('}');
+				return object; // empty object
+			}
+			while (this.ch) {
+				if (this.ch >= "0" && this.ch <= "9") {
+					key = this.number();
+				} else if (this.ch === '"' || this.ch === '\'') {
+					key = this.string();
+				} else {
+					key = this.name();
+				}
+				this.white();
+				this.next(':');
+				if (Object.hasOwnProperty.call(object, key)) {
+					this.error('Duplicate key "' + key + '"');
+				}
+				object[key] = this.value();
+				this.white();
+				if (this.ch === '}') {
+					this.next('}');
+					return object;
+				}
+				this.next(',');
+				this.white();
+			}
+		}
+		this.error("Bad object");
+	};
+
+	JSTokenizer.prototype.value = function() {
+
+		// Parse a JS value. It could be an object, an array, a string, a number,
+		// or a word.
+		this.white();
+		switch (this.ch) {
+			case '{':
+				return object.call(this);
+			case '[':
+				return this.array();
+			case '"':
+			case '\'':
+				return this.string();
+			case '-':
+				return this.number();
+			default:
+				return this.ch >= '0' && this.ch <= '9' ? this.number() : this.word();
+		}
+	};
+
+	// /**
+	//  * Returns the index of the current character.
+	//  *
+	//  * @private
+	//  * @returns {int} The current character's index.
+	//  */
+	JSTokenizer.prototype.getIndex = function() {
+		return this.at - 1;
+	};
+
+	JSTokenizer.prototype.getCh = function() {
+		return this.ch;
+	};
+
+	JSTokenizer.prototype.init = function(sSource, iIndex) {
+		this.text = sSource;
+		this.at = iIndex || 0;
+		this.ch = ' ';
+	};
+
+	// /**
+	//  * Advances the index in the text to <code>iIndex</code>. Fails if the new index
+	//  * is smaller than the previous index.
+	//  *
+	//  * @private
+	//  * @param {int} iIndex - the new index
+	//  */
+	JSTokenizer.prototype.setIndex = function(iIndex) {
+		if (iIndex < this.at - 1) {
+			throw new Error("Must not set index " + iIndex
+				+ " before previous index " + (this.at - 1));
+		}
+		this.at = iIndex;
+		this.next();
+	};
+
+	// /**
```

Comment:
Wouldn't it be possible to exclude the file from the JSDoc build of the tooling? At least this would simplify regular updates...

Well, the modification of the module definition remains necessary anyhow.

Reply:
Yes, that's right.
I mainly wanted to get `npm test` working and discuss the re-use/integration of JSTokenizer beforehand.

Reply:
Done